rllib/algorithms/apex_dqn/apex_dqn.py

```diff
@@ -593,7 +593,7 @@ def wait_on_replay_actors() -> None:
                 # the minimum threshold yet.
                 if item:
                     self.learner_thread.inqueue.put(
-                        self.replay_sample_batches[0], timeout=0.001
+                        self.replay_sample_batches[0], block=True
```

Comment:
hey man, can you like write a lot of comments detailing the journey you went through and all the finding, so we know why we made the choice to block here, which not at all intuitive?
thanks.

Reply:
yeah :)